### DISCUSSION TITLE
Issue with unusual table partition name

### DISCUSSION THREAD

**Discussion by benperry-ucas** [sqlalchemy/sqlalchemy#10901] [Usage Questions]
Hi all,

I was wondering if anyone could help me with an issue with unusual names for partitions used by my organisation. I think the issue is similar to to the issue below.

https://github.com/sqlalchemy/sqlalchemy/issues/10275

I added a couple of print statements to the get_multi_foreign_keys method and saw the following:

![image](https://github.com/sqlalchemy/sqlalchemy/assets/157111160/f7a4d66f-710f-4bb5-b048-192f7425900d)

So it looks like the issue is with the "dim_application_partition_-1" name. Is it the '-' causing the problem?

It's worth noting that downgrading to version 1.4 fixes the issue.

Any help would be greatly appreciated.

**Comment by CaselIT** [comment]
Hi,

Created an issue since this is likely a bug https://github.com/sqlalchemy/sqlalchemy/issues/10902
It seems that the regular expression does not account for that form.

PR are welcome (if you need direction on where to add test feel free to ask!)
